channel.go

```diff
@@ -191,11 +202,20 @@ func NewChannel(serviceName string, opts *ChannelOptions) (*Channel, error) {
 		relayStats = opts.RelayStats
 	}
 
+	maxMillis := opts.RelayMaxTimeout / time.Millisecond
+	if opts.RelayMaxTimeout == 0 {
+		opts.RelayMaxTimeout = defaultRelayMaxTimeout
+	} else if opts.RelayMaxTimeout < 0 || maxMillis > math.MaxUint32 {
+		logger.WithFields(
+			LogField{"configuredMaxTimeout", opts.RelayMaxTimeout},
+			LogField{"defaultMaxTimeout", defaultRelayMaxTimeout},
+		).Warn("Configured RelayMaxTimeout is invalid, using default instead.")
+		opts.RelayMaxTimeout = defaultRelayMaxTimeout
```

Comment:
Considered it, but I really don't like storing durations as anything other than a `time.Duration`. Even in the hot path, it doesn't seem worth it to save 0.3 ns.